### ISSUE TITLE
Last value is missing

### ISSUE THREAD

**Issue by joemoe** [raintank/crate-datasource#10]
When I do a hourly aggregation and there is "part of an hour left" it stops the graphs on the right.

As you can see here.

<img width="285" alt="bildschirmfoto 2016-08-04 um 17 24 19" src="https://cloud.githubusercontent.com/assets/68620/17408047/ef3dd8c6-5a69-11e6-9a68-0765cf550419.png">

**Comment by alexanderzobnin** [comment]
Can you show me your query?

**Comment by alexanderzobnin** [comment]
it's due `date_trunc('minute'...)`. So all points in each minute has been aggregated into one with rounded timestamp.

**Comment by jodok** [comment]
@alexanderzobnin how do other datasources deal with that issue?

**Comment by alexanderzobnin** [comment]
InfluxDB example (group by 1 hour):

<img width="1280" alt="screen shot 2016-09-22 at 20 55 30" src="https://cloud.githubusercontent.com/assets/4932851/18759793/ec87a67a-8106-11e6-8e31-5a552854ebec.png">

**Comment by jodok** [comment]
ok, so let's leave this issue with low prio as well. seems to be a general, cosmetic issue

**Comment by alexanderzobnin** [comment]
Seems, `GROUP BY` from both Crate and InfluxDB uses left bound's timestamp for grouped value.

**Comment by jodok** [comment]
i think the only possibility is that you make sure when building the WHERE clause  that the upper limit is high enough.

**Comment by joemoe** [comment]
yes, if you aggregate by a minute the right bound should be the next full minute or - which might be even easier - round the shown timeframe to the time interval.
